utility/split-log-into-stages.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e
+set -u
+IFS=$'\n'
+
+cd "$(dirname "$1")"
+
+full_log_gz=$(basename "$1")
+gunzip --keep "$full_log_gz"
+full_log=$(basename "$1" .gz)
+
+function get_stage_regexp() {
+    echo '^.{29}(?:\[Pipeline\] )?\[(?!Pipeline)('"$1"')\].*$'
+}
+
+stage_regexp=$(get_stage_regexp '[\w -]+')
+
+stages=$(perl -lne "s/$stage_regexp/\$1/ or next; print" < "$full_log" | sort --unique)
+
+for stage in $stages
+do
+    stage_slug=$(echo "$stage" | tr '[:upper:]' '[:lower:]' | tr --squeeze ' ' '-')
+    stage_log_filename="log.$stage_slug.txt.gz"
+    this_stage_regexp=$(get_stage_regexp "$stage")
+    echo "copying '$stage' log to $stage_log_filename"
+    grep --perl-regexp "$this_stage_regexp" "$full_log" | gzip > "$stage_log_filename"
+done
+
+non_tagged_filename='log.cloning-tests-and-post.txt.gz'
```

Comment:
What do you think about naming it `cloning-and-tests-and-post` instead of `cloning-tests-and-post`?
Someone, somewhere might be thinking this log is dedicated to "cloning test", not "cloning" and "testing" separately.